components/date_picker/date_picker.tsx

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useRef, useState} from 'react';
+
+import FocusTrap from 'focus-trap-react';
+import {DayPicker, DayPickerProps} from 'react-day-picker';
+import {usePopper} from 'react-popper';
+
+import type {Locale} from 'date-fns';
+
+import 'react-day-picker/dist/style.css';
+import './date_picker.scss';
+
+type Props = {
+    children: React.ReactNode;
+    datePickerProps: DayPickerProps;
+    isPopperOpen: boolean;
+    locale: string;
+    closePopper: () => void;
+}
+
+const DatePicker = ({children, datePickerProps, isPopperOpen, closePopper, locale}: Props) => {
+    const loadedLocales: Record<string, Locale> = {};
+    const popperRef = useRef<HTMLDivElement>(null);
+    const [popperElement, setPopperElement] = useState<HTMLDivElement | null>(
+        null,
+    );
+
+    const popper = usePopper(popperRef.current, popperElement, {
+        placement: 'bottom-start',
+    });
+
+    const localeExists = (path: string) => {
+        try {
+            require.resolve(path);
+            return true;
+        } catch (e) {
+            return false;
+        }
+    };
+
+    useEffect(() => {
+        if (isPopperOpen && !popperRef?.current) {
+            closePopper();
+        }
+    }, [popperRef]);
+
+    useEffect(() => {
+        if (locale && locale !== 'en-US' && !loadedLocales[locale] && localeExists(`date-fns/locale/${locale}/index.js`)) {
+            /* eslint-disable global-require */
+            loadedLocales[locale] = require(`date-fns/locale/${locale}/index.js`);
```

Comment:
The reason to import them like this is because we need to dynamically import locales because we don't know what locale the user has beforehand. There is no way to import all locales at once and unfortunately we can't use moment with this package